torch/_tensor_docs.py

```diff
@@ -513,7 +513,17 @@ def add_docstr_all(method, docstr):
                r"""
 logical_not_() -> Tensor
 
-In-place version of :meth:`~Tensor.logical_not`
+In-place version of :meth:`~Tensor.logical_not`, except that it always returns a tensor with the same dtype instead of
```

Comment:
I don't think we want to document this separately -- it's too difficult to maintain.  The alternative dtype should be covered by the `out` argument for the non-inplace method.